cli/command/cluster/health.go

```diff
@@ -62,81 +65,49 @@ func runHealth(storageosCli *command.StorageOSCli, opt *healthOpt) error {
 		}
 	}
 
-	if err := cliTypes.SortCLINodes(cliTypes.ByNodeName, nodes); err != nil {
-		return err
-	}
-
-	for _, node := range nodes {
-		if err := runNodeHealth(node, opt.timeout); err != nil {
-			return err
-		}
-	}
-
 	clusterHealthCtx := formatter.Context{
 		Output: storageosCli.Out(),
 		Format: formatter.NewClusterHealthFormat(format, opt.quiet),
 	}
-	return formatter.ClusterHealthWrite(clusterHealthCtx, nodes)
-}
-
-func runNodeHealth(node *cliTypes.Node, timeout int) error {
-	addr := node.AdvertiseAddress
-
-	u, err := url.Parse(node.AdvertiseAddress)
-	if err == nil && u.Host != "" {
-		addr = u.Host
+	if err := formatter.ClusterHealthWrite(clusterHealthCtx, status); err != nil {
+		return err
+	} else if opt.errReturn {
+		for _, s := range status {
+			// Check for all the submodules that need to be alive for us to be
+			// functioning
+			if err := checkSubmodules(s); err != nil {
+				return err
+			}
+		}
 	}
-
-	cliNode.UpdateNodeHealth(node, addr, timeout)
-
 	return nil
 }
 
-func getNodes(storageosCli *command.StorageOSCli, opt *healthOpt) ([]*cliTypes.Node, error) {
-
-	if opt.cluster != "" {
-		return getDiscoveryNodes(storageosCli.GetDiscovery(), opt.cluster)
+func checkSubmodules(nodeSubmodules *apiTypes.ClusterHealthNode) error {
+	const desiredStatus = "alive"
+	const errTemplate = "\nnode: %s has unhealthy submodues:\n%s"
+	const submoduleTemplate = "name: %s, message: '%s'"
```

Comment:
`%q` is like`"%s"` except it will escape any stray quotes in the string being expanded 👍

Reply:
Fixed